optuna/storages/rdb/storage.py

```diff
@@ -477,9 +500,11 @@ def _merge_trials_orm(
             for system_attr in id_to_system_attrs[trial_id]:
                 system_attrs[system_attr.key] = json.loads(system_attr.value_json)
 
+            # TODO(Yanase): Use trial.number after TrialModel.number is added.
+            trial_number = system_attrs.pop('number', trial_id)
```

Comment:
I think this line should be: `trial_number = system_attrs.pop('number', self.get_trial_number_from_id(trial_id))`

Reply:
As our offline discussion, I'll keep `number` in FrozenTrial, and rename it to `_number`.